File: examples_ios/MyApp/Sources/Controllers/ViewModels/Models/ProgramModel.swift
import ConfirmitMobileSDK
import Foundation

class ProgramModel {
  private(set) var program: Program
  private(set) var configs: ProgramConfig

  init(program: Program) throws {
    self.program = program
    configs = ProgramConfig(serverId: program.serverId, programKey: program.programKey)
  }

  func update(program: Program) throws {
    self.program = program
  }

  func getCustomDataValues() -> [String: String] {
    var values = [String: String]()
    let raw = configs.customData
    for pair in raw.split(separator: ";") {
      let keyValue = pair.split(separator: "=")
      if keyValue.count == 2 {
        let key = String(keyValue[0])
        let value = String(keyValue[1])
        values[key] = value
      }
    }
    return values
  }
}

File: examples_android/app/src/main/java/com/example/myapp/fragments/viewmodels/models/ProgramModel.kt
package com.example.myapp.fragments.viewmodels.models

import androidx.annotation.Keep
import com.confirmit.mobilesdk.database.externals.Program
import com.example.myapp.domain.ProgramConfig
import java.io.Serializable

@Keep
class ProgramModel(program: Program) : Serializable {
    var program: Program = program
        private set
    var configs: ProgramConfig
        private set

    init {
        this.configs = ProgramConfig(program.serverId, program.programKey)
    }

    fun update(program: Program) {
        this.program = program
    }

    fun getCustomData(): Map<String, String> {
        val values = mutableMapOf<String, String>()
        val raw = configs.customData
        raw.split(";").forEach {
            val keyValue = it.split("=")
            if (keyValue.size == 2) {
                val key = keyValue[0]
                val value = keyValue[1]
                values[key] = value
            }
        }
        return values
    }
}